light_controller/ir_toy/ir_toy.py

```diff
@@ -0,0 +1,352 @@
+
+    def writeArray(self, data):
+        """
+        Write a list of bytes
+        :param data: list of bytes
+        :return: number of bytes written
+        """
+        bytes = bytearray(data)
+        return self.write(bytes)
+
+    def writeByte(self, data):
+        """
+        Convert a byte to char and write it out
+        :param data: byte to write
+        :return: number of bytes written
+        """
+        return self.writeArray([data, ])
+
+    def enableTransmitMode(self):
+        """
+        Set transmission mode
+        :return: Size of TX buffer
+        """
+        self.writeByte(CommandEnum.transmit)
+        return ord(self.read(1))
+
+    def getFrequencyReport(self):
+        """
+        Get the raw timing information for the previous IR signal
+
+        :return: (timer at 2nd rising edge,
+                  timer at 3rd rising edge,
+                  timer at 4th rising edge,
+                  total count of pulses since last 0xFF)
+        """
+        self.writeByte(CommandEnum.getFrequencyReport)
+        buffer = self.read(8)
+        return struct.unpack(">4H", buffer)
+
+    def setSampleTimer(self, prescale):
+        """
+        The timer is 21.13333 us by default.
+        111 (0x07)	1:256	21.3333uS
+        110 (0x06)	1:128	10.6666uS
+        101 (0x05)	1:64
+        100 (0x04)	1:32
+        011 (0x03)	1:16
+        010 (0x02)	1:8
+        001 (0x01)	1:4
+        000 (0x00)	1:2
+        :param prescale: see above for valid values
+        :return: None
+        """
+        self.writeArray((CommandEnum.setSampleTimer, prescale))
+
+    def setTX_Modulation(self, pr2, dutyCycle=0x0):
+        """
+        Configures the TX modulation of the IR Transmitter
+        IR sample mode starts with a default 36kHz infrared transmit modulation frequency. The rate can be changed
+        with the setup modulation command, followed by a byte that sets the pulse-width modulator match value (PR2),
+        and a don't care byte.
+        :param pr2: Pulse width modulator match value
+        :param dutyCycle: Duty Cycle
+        :return: None
+        """
+        self.writeArray((CommandEnum.setFrequencyModulationTimer, pr2, dutyCycle))
+
+    def setMuteLED_On(self):
+        """
+        Disables the LED during sample mode
+        :return: None
+        """
+        self.writeByte(CommandEnum.setLED_MuteOn)
+
+    def setMuteLED_Off(self):
+        """
+        Enables the LED during sample mode
+        :return: None
+        """
+        self.writeByte(CommandEnum.setLED_MuteOff)
+
+    def setLED_On(self):
+        """
+        Turn the LED on
+        :return: None
+        """
+        self.writeByte(CommandEnum.setLED_On)
+
+    def setLED_Off(self):
+        """
+        Turn the LED off
+        :return: None
+        """
+        self.writeByte(CommandEnum.setLED_Off)
+
+    def getSettings(self):
+        """
+        Returns 8 bytes that describe the current IR Toy settings:
```

Comment:
This description doesn't seem to be correct as we are returning an IR_ToySettings instead.